tests/apps/btc_app/btc_helpers_tests.c

```diff
@@ -0,0 +1,108 @@
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+#include "btc_app.h"
+#include "btc_helpers.h"
+#include "btc_priv.h"
+#include "curves.h"
+#include "flash_config.h"
+#include "unity_fixture.h"
+#include "utils.h"
+
+TEST_GROUP(btc_helper_test);
+
+/**
+ * @brief Test setup for usb event consumer tests.
+ * @details The function populates data in local buffer of USB communication
+ * module so that the event getter has an event ready for dispatch for
+ * performing tests. buffer of packet(s) of data.
+ */
+TEST_SETUP(btc_helper_test) {
+}
+
+/**
+ * @brief Tear down the old test data
+ * @details The function will perform cleanup of the current running test and
+ * bring the state of execution to a fresh start. This is done by using purge
+ * api of usb-event and clearing buffers using usb-comm APIs.
+ */
+TEST_TEAR_DOWN(btc_helper_test) {
+  g_app = NULL;
+}
+
+TEST(btc_helper_test, btc_txn_helper_xpub_btc) {
+  uint32_t path[] = {0x8000002C, 0x80000000, 0x80000000};
+  char xpub[XPUB_SIZE] = "";
+  char expected_xpub[] =
+      "xpub6CmEVMqHxkaGqdu3zxNRTx3fuExwg95T3HN3dY5t9oi3sLYzVupt6HTJXXZGWVjBQTwD"
+      "7ZJi6HyM8BxxUhF8aqA14PBkzX6su56szCmxJ5o";
+  uint8_t seed[64] = {0};
+
+  hex_string_to_byte_array(
+      "000000000000000000000000000000000000000000000000000000000000000000000000"
+      "00000000000000000000000000000000000000000000000000000000",
+      128,
+      seed);
+  btc_generate_xpub(path,
+                    sizeof(path) / sizeof(uint32_t),
+                    SECP256K1_NAME,
+                    seed,
+                    get_btc_app()->legacy_xpub_ver,
+                    xpub);
```

Comment:
Please re-check if this test is required in this PR

Reply:
this got duplicated. removed.